metakb/deltas/civic.py

```diff
@@ -0,0 +1,139 @@
+"""A module for computing CIViC deltas."""
+from metakb import PROJECT_ROOT
+import json
+import logging
+from jsondiff import diff
+from datetime import date
+import pkg_resources
+from metakb.harvesters import CIViC
+
+logger = logging.getLogger('metakb')
+logger.setLevel(logging.DEBUG)
+
+
+class CIVICDelta:
+    """A class for computing CIViC deltas."""
+
+    def __init__(self, main_json, *args, **kwargs):
+        """Initialize the CIVICDelta class.
+
+        :param str main_json: The path to the main CIViC composite json file.
+        """
+        self._main_json = main_json
+        if '_new_json' in kwargs:
+            self._new_json = kwargs['_new_json']
+        else:
+            self._new_json = None
+
+    def compute_delta(self):
+        """Compute delta for CIViC and store computed delta in a JSON file.
+
+        :return: A dictionary of ids to delete, update, or insert to the main
+                 harvester.
+        """
+        # Main harvester
+        with open(self._main_json, 'r') as f:
+            main_civic = json.load(f)
+
+        current_date = date.today().strftime('%Y%m%d')
+
+        # New harvester
+        if self._new_json:
+            with open(self._new_json, 'r') as f:
+                new_civic = json.load(f)
+        else:
+            fn = f"civic_harvester_{current_date}.json"
+            c = CIViC()
+            c.harvest(fn=fn)
+
+            with open(f"{PROJECT_ROOT}/data/civic/{fn}", 'r') as f:
+                new_civic = json.load(f)
+
+        delta = {
+            '_meta': {
+                'civicpy_version':
+                    pkg_resources.get_distribution("civicpy").version,
+                # TODO: Might change. Assuming we harvest when computing delta
```

Comment:
More broadly, we should think about what metadata to provide with harvested information (and how delta computation/update methods modify or augment it)

Reply:
I agree. I don't believe CIViCpy includes metadata at the moment. I know @bwalsh mentioned keeping track of the tools and versions, so that's why I included lines 54-55. I believe the lines below (which are not shown) include a `date_harvested` field. What other fields were you thinking of including @jsstevenson ?

Reply:
I added `metakb_version`. I'm unsure if we need `civicpy_version`.